File: server/src/main/java/org/dan/jadalnia/sys/type/number/ImmutableNumber.java
package org.dan.jadalnia.sys.type.number;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonValue;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@EqualsAndHashCode
@RequiredArgsConstructor(onConstructor = @__(@JsonCreator))
public class ImmutableNumber extends AbstractNumber {
    private final int value;

    @Override
    @JsonValue
    public int intValue() {
        return value;
    }

    @JsonIgnore
    public int getValidateValue() {
        return value;
    }
}

File: kot/src/main/kotlin/org/dan/jadalnia/sys/type/number/ImmutableNumber.kt
package org.dan.jadalnia.sys.type.number;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonValue;

open class ImmutableNumber
@JsonCreator
constructor (val value: Int)
    : AbstractNum() {
    @Override
    @JsonValue
    override fun intValue(): Int = value

    @JsonIgnore
    open fun getValidateValue(): Int = value

    override fun hashCode(): Int = value
    override fun equals(other: Any?): Boolean {
        if (other is AbstractNum) {
            return value == other.intValue()
        }
        return false
    }
}
